### PR TITLE
add AdaGrad optimizer in python

### PR DESCRIPTION
**Pull request by linmx0130**
A simple implementation of AdaGrad optimizer in Python

### CODE REVIEW

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -464,6 +464,45 @@ def update(self, index, weight, grad, state):
...
+        L2 regularization coefficient add to all the weights
+    rescale_grad : float, optional
+        rescaling factor of gradient.
+    eps: float, optional
+        A small float number to make the updating processing stable
+        Default value is set to 1e-7.
+    """
+    def __init__(self, learning_rate=0.05, wd=0. , rescale_grad=1, eps = 1e-7, arg_names= None):
+        super(AdaGrad, self).__init__(rescale_grad, arg_names, wd)
+        self.lr=learning_rate
```
Comment (winstywang) [inline python/mxnet/optimizer.py]: Pay attention to spacing before and after =

File: python/mxnet/optimizer.py [hunk 2]
```diff
@@ -464,6 +464,45 @@ def update(self, index, weight, grad, state):
...
+        self.rescale_grad= rescale_grad
+    def create_state(self, index, weight):
+        return (zeros(weight.shape, weight.context),   #history
+                zeros(weight.shape, weight.context))   #delta
+    
+    def update(self, index, weight, grad, state):
+        assert(isinstance(weight, NDArray))
+        assert(isinstance(grad, NDArray))
+        grad= grad*self.rescale_grad
+        history,delta = state
```
Comment (winstywang) [inline python/mxnet/optimizer.py]: Spacing after ,

Comment (sxjscience) [inline python/mxnet/optimizer.py]: It seems unnecessary to keep two states. AdaGrad will work fine using only the `history` state.

Comment (linmx0130) [inline python/mxnet/optimizer.py]: I don't know how to update the code in pull&request so I open a new pull request  #1580 . Thanks for attention! @winstywang @sxjscience

Comment (sxjscience) [inline python/mxnet/optimizer.py]: @linmx0130 You can just push your local changes of the PR branch to the remote and github will update this page and run tests automatically.

**Pull request closed without merging**